src/frontend/js/components/PurchaseButton/index.tsx

```diff
@@ -42,6 +42,10 @@ const messages = defineMessages({
 interface PurchaseButtonProps {
   product: Joanie.Product;
   course: Joanie.CourseLight;
+  // If the product is a certificate, the courseRun is required.
+  courseRun?: Joanie.CourseRun;
```

Comment:
Maybe out of scope of this PR ?
Furthermore, in case of Certificate product, you will have to pass an Enrollment so we are able to access to CourseRun from the enrollment.